hie-compat/src-ghc86/Compat/HieBin.hs

```diff
@@ -327,6 +327,9 @@ putName (HieSymbolTable next ref) bh name = do
 
 -- ** Converting to and from `HieName`'s
 
+{- HLINT ignore toHieName "Avoid restricted function" -}
+
+-- | Panics if the name is of sort Internal or System.
 toHieName :: Name -> HieName
 toHieName name
   | isKnownKeyName name = KnownKeyName (nameUnique name)
```

Comment:
nameModule vs nameModule_maybe. Presumably we trust this usage since it's here in the code and working.

Personally I would favor defining and using (unsafeFromJust $ nameModule_maybe name). Or unsafeNameModule.